### ISSUE TITLE
Splitting the data is wrong !?

### ISSUE THREAD

**Issue by Amrusama** [TsingZ0/PFLlib#194]
I generated a non-iid version of FashionMINST  for 15 clients using the following command 
`python generate_FashionMNIST.py noniid - pat`
The output of the data distribution on the terminal is as follows: 

![Screenshot 2024-07-02 141328](https://github.com/TsingZ0/PFLlib/assets/61755893/639b7c58-ead0-4b74-a988-e03e94435e0a)

I printed the number of data points for each class in every client and I plotted the distribution of every client class and it didn't match the output on the terminal.

![Screenshot 2024-07-02 140351](https://github.com/TsingZ0/PFLlib/assets/61755893/b7623c2b-3167-4894-af84-6117e05d8c24)
![Screenshot 2024-07-02 140116](https://github.com/TsingZ0/PFLlib/assets/61755893/138d4b27-adc2-4fa6-ae8a-3ba896e68efa)

**Comment by TsingZ0** [comment]
Please differentiate between the "entire set" and the "training set." The training set is almost 75% of the entire set for each client.

**Comment by Amrusama** [comment]
@TsingZ0 Thank you for the explanation. I suggest including this information in the dataset generation section. For instance, the entire training set of FashionMNIST comprises 70,000 samples. My intention was for PFLib to provide a non-iid version of the entire training set.

**Comment by TsingZ0** [comment]
The test set can be reshuffled after it has been split.
